src/components/widgets/filesystem/FileSystem.vue

```diff
@@ -437,13 +437,18 @@ export default class FileSystem extends Mixins(StateMixin, FilesMixin, ServicesM
         // Clear selected bulk items if we're navigating folders.
         this.selected = []
       } else {
-        // Open the context menu
-        this.contextMenuState.x = e.clientX
-        this.contextMenuState.y = e.clientY
-        this.contextMenuState.file = item
-        this.$nextTick(() => {
-          this.contextMenuState.open = true
-        })
+        if (e.type === 'click' && item.type === 'file' && ['txt', 'md', 'conf', 'cfg', 'ini', 'sh'].includes(item.extension)) {
+          // Open the file editor
+          this.handleFileOpenDialog(item as AppFile)
```

Comment:
The `as AppFile` is not needed here, TypeScript should infer that automatically from the above `type === 'file'` check!